### ISSUE TITLE
Hunk and line staging

### ISSUE THREAD

**Issue by franciscolourenco** [jesseduffield/lazygit#46]
Is there support for partial file staging? Like hunks or lines?

**Comment by jesseduffield** [comment]
Currently no, as this is blocked by https://github.com/jesseduffield/lazygit/pull/18
But it's the number 1 feature I want in the app, particularly for personal use!
Closing as this is a duplicate of https://github.com/jesseduffield/lazygit/issues/11

**Comment by Flimm** [comment]
For the lukers, newer versions of Lazygit support this feature. To use it, press <kbd>2</kbd> to focus on the list of files, select the file you wish to focus on, and press <kbd>enter</kbd>. Navigate through the file using the arrow keys, and press <kbd>space</kbd> for every line that you wish to stage. You can switch between the unstaged file and the staged file using the <kbd>tab</kbd> key. When you're ready, press <kbd>esc</kbd>, then <kbd>c</kbd> to commit the staged changes.

**Comment by louwers** [comment]
And you can press `a` to toggle between staging hunks and staging line by line.
